### PR TITLE
[Added] [BEL-10869] Add support for tax retentions endpoint

### PR DESCRIPTION
**Pull request by GeoffreyVanderCruyssen**
Adding support and tests for the recently added `/api/tax-retentions/` endpoint

### CODE REVIEW

File: tests/taxRetentions.test.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
+import 'regenerator-runtime/runtime';
```
Comment (GeoffreyVanderCruyssen) [inline tests/taxRetentions.test.js:1]: I had to add this line specifically to make this test suite work, while nothing is different from the other test files. Not sure why, this is an inconsistency that we should take a look at.

Comment (guimarquesbelvo) [inline tests/taxRetentions.test.js:1]:
```suggestion
import nock from 'nock'; 
import { APIMocker, newSession } from './fixtures';
import TaxRetentions from '../src/taxRetentions';
```

I tested swapping these lines here in other tests and the same error appears. This should do the fix

**Review by guimarquesbelvo — DISMISSED** [review]

File: tests/taxRetentions.test.js (hunk 1 above)
Comment (GeoffreyVanderCruyssen) [reply tests/taxRetentions.test.js:1]: image

File: tests/taxRetentions.test.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  collected_at: '2022-02-09T08:45:50.406032Z',
+  created_at: '2022-02-09T08:46:20.406032Z',
+  invoice_identification: 'def404af-5eef-4112-aa99-d1ec8493b89a',
+  version: '1.0',
+  code: 25,
+  issued_at: '2019-01-03T21:10:40.000Z',
+  certified_at: '2019-01-03T21:10:41.000Z',
+  cancelled_at: null,
+  sender_id: 'JKUF980404P0',
+  sender_name: 'Roberto Nunez Batman',
```
Comment (sergio-belvo) [inline tests/taxRetentions.test.js:17]: 🦇 👨

**Review by sergio-belvo — APPROVED** [review]
🚀

File: src/taxRetentions.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+class TaxRetentions extends Resource {
+  #endpoint = 'api/tax-retentions/'
+
+  /**
+   * Retrieve tax retentions information from a specific fiscal link.
+   *
+   * @async
+   * @param {string} link - UUID4 representation of a link Id.
+   * @param {string} DateFrom - Required date from, format is YYYY-MM-DD
```
Comment (benitocas88) [inline src/taxRetentions.js:11]: Ey @GeoffreyVanderCruyssen note that params documented must match with function args, in this case `dateFrom` ans `dateTo`

File: tests/taxRetentions.test.js (hunk 2 above)
Comment (benitocas88) [reply tests/taxRetentions.test.js:17]: 🦇

**Review by benitocas88 — APPROVED** [review]
Probably out of the scope but in case you want, you can implement and test `notImplementError` (custom throw in js) as we do in belvo-python https://github.com/belvo-finance/belvo-python/blob/master/belvo/resources/tax_retentions.py#L54

File: src/taxRetentions.js (hunk 3 above)
Comment (GeoffreyVanderCruyssen) [reply src/taxRetentions.js:11]: @benitocas88 I just looked at what was in [invoices here](https://github.com/belvo-finance/belvo-js/blob/master/src/invoices.js#L1-L44)

File: src/taxRetentions.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import Resource from './resources';
+
+class TaxRetentions extends Resource {
```
Comment (AlbertMaristany) [inline src/taxRetentions.js:3]: We need some documentation on top of the resource. You can check other resources or ask some help to our docs overlord @dom-techwriter

**Review by AlbertMaristany — CHANGES_REQUESTED** [review]
:muscle: Looking good! Lets add some documentation and move on.

File: src/taxRetentions.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import Resource from './resources';
+
+/**
+ * A Tax retention is HI DOM, HELP A STUPID OUT
+ *
+ * @extends Resource
+ */
```
Comment (GeoffreyVanderCruyssen) [inline src/taxRetentions.js:7]: Hi @dom-techwriter , how are you?

File: src/taxRetentions.js (hunk 4 above)
Comment (blv-dominik) [reply src/taxRetentions.js:10]: This looks good to me! @AlbertMaristany and @GeoffreyVanderCruyssen

**Review by blv-dominik — APPROVED** [review]
looks good to me from a docs perspective! :)

**Review by AlbertMaristany — APPROVED** [review]

**Pull request merged**